### ISSUE TITLE
Multi-labels

### ISSUE THREAD

**Issue by lucamarini22** [KevinMusgrave/pytorch-metric-learning#742]
Hello! First of all, thanks for this awesome library.

Is there a way to pass to a loss function a matrix of 0 and 1 (or False and True) of every possible pair instead of passing the class labels?

I'm asking this to manage the case where a sample belongs to multiple classes.

**Comment by KevinMusgrave** [comment]
For contrastive losses (ContrastiveLoss, NTXentLoss etc.) you can omit the labels and pass in `indices_tuple`:

```python
loss_fn = ContrastiveLoss()
loss = loss_fn(embeddings, indices_tuple=indices_tuple)
```

Here's an example of how you can specify positive and negative pairs using `indices_tuple`: https://github.com/KevinMusgrave/pytorch-metric-learning/discussions/435#discussioncomment-2268101

The discussion in the link is about the output of a miner, but you don't have to use a miner. You can manually create the `indices_tuple` yourself.

**Comment by lucamarini22** [comment]
Great, thanks!

**Comment by lucamarini22** [comment]
For people that might be interested: for the full list of losses that can use `indices_tuples`, see [here](https://kevinmusgrave.github.io/pytorch-metric-learning/losses/) under "Losses for which you can pass in indices_tuple without labels"

**Comment by lucamarini22** [comment]
Hi! Is there a way to pass the indices_tuples directly to the miner as well?

**Comment by KevinMusgrave** [comment]
No unfortunately there isn't. What behavior would you like when passing `indices_tuple` to a miner?

**Comment by lucamarini22** [comment]
I'd like to select the hard_pairs based on the `indices_tuple` as in my case I am also using `indices_tuple` when computing the loss, as I don't have classes that are categories. Does this clarify the expected behaviour? Please let me know in case you have other questions

**Comment by lucamarini22** [comment]
Actually I think I found a way to make it work, I will open a PR when I will have some time to clean the code!
